### DISCUSSION TITLE
Clarification regarding documentation vs implementation of Authors/Maintainers

### DISCUSSION THREAD

**Discussion by blueninja416** [python-poetry/poetry#9349] [Q&A] [answered]
In the documentation for the `pyproject.toml` file, it says that you can include multiple authors/maintainers by adding them as extra items into their respective fields like below:

```toml
maintainers = ["user1 <user1@gmail.com>", "user2 <user2@gmail.com>"]
```

However, this only includes the first item in that list. Is this the expected behavior? If so, I think this should be clarified in the documentation. I saw from a couple issues that the core python metadata only supports one of each, however it does support multiple users/emails in the `author-email`/`maintainer-email` fields so if it's a bug in the implementation, is it supposed to add those values in those fields instead?

**Comment by radoering** [comment]
See  #7229, especially https://github.com/python-poetry/poetry/issues/7229#issuecomment-1360017235.

On the other side there is https://packaging.python.org/en/latest/specifications/pyproject-toml/#authors-maintainers which gives some instructions how to fill core metadata so maybe we should implement this.

PS: If anyone knows a package with multiple values in its core metadata, I'd like to see how PyPI displays the authors/maintainers.

**Reply by blueninja416** [reply]
Looks like PyPI only displays the first maintainer as well, with this package at least: https://pypi.org/project/llama-index-cli/

**Reply by blueninja416** [reply]
There's also https://pypi.org/project/histolab/ which displays multiple maintainers in a list.

**Reply by radoering** [reply]
That's because there is only one maintainer if you download the wheel and open the `METADATA` file in the `.dist-info` folder:

```
Maintainer: Andrei Fajardo
Maintainer-email: andrei@runllama.ai
```

Same for `PKG-INFO` in the sdist.

What I'm interested in is what PyPI displays if there where multiple comma separated values in one of the fields. (Since we know that Poetry does not put multiple values there, it has to be a package that is built with another build backend.)

**Reply by radoering** [reply]
> There's also https://pypi.org/project/histolab/ which displays multiple maintainers in a list.

That's better. Now we only need an example for names + e-mails.

**Reply by blueninja416** [reply]
Looks like that example does use names + emails: https://github.com/histolab/histolab/blob/78c869c4a501c634a811d102d7064c987dc9cae1/pyproject.toml#L8

**Reply by radoering** [reply]
Yeah, but it uses Poetry and a list of maintainers, so only the first value is in the `METADATA` file.

**Comment by radoering** [comment, marked as answer]
Found https://github.com/pypi/warehouse/issues/9400

It seems like PyPI will not display authors/maintainers correctly if we comply with the standard. Since PyPI is big enough to change the standard it might be better to wait for a decision from their side...
